app/connectors/ArrivalMovementConnector.scala

```diff
@@ -18,16 +18,32 @@ package connectors
 
 import config.FrontendAppConfig
 import javax.inject.Inject
-import models.Arrivals
+import models.{ArrivalId, Arrivals}
+import play.api.libs.ws.WSClient
 import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.http.HttpClient
+import cats.implicits._
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class ArrivalMovementConnector @Inject()(config: FrontendAppConfig, http: HttpClient)(implicit ec: ExecutionContext) {
+class ArrivalMovementConnector @Inject()(config: FrontendAppConfig, http: HttpClient, ws: WSClient)(implicit ec: ExecutionContext) {
 
   def getArrivals()(implicit hc: HeaderCarrier): Future[Arrivals] = {
     val serviceUrl: String = s"${config.destinationUrl}/movements/arrivals"
     http.GET[Arrivals](serviceUrl)
   }
+
+  def getPDF(arrivalId: ArrivalId)(implicit hc: HeaderCarrier): Future[Option[Array[Byte]]] = {
+    val serviceUrl: String = s"${config.destinationUrl}/movements/arrivals/${arrivalId.index}/unloading-permission"
+
+    hc.authorization.traverse(
```

Comment:
Rather than pass the header carrier here, can we pass the Authorization token here, then we can change this to be a 
`Future[Array[Byte]]`? We know that user _should_  have a authorization token at this point but because this layer is messing around with the HeaderCarrier we're forced to have the option here